api/src/auth/drivers/openid.ts

```diff
@@ -94,36 +80,78 @@ export class OpenIDAuthDriver extends LocalAuthDriver {
 			throw new InvalidProviderError();
 		}
 
-		this.client = new Promise((resolve, reject) => {
-			Issuer.discover(issuerUrl)
-				.then((issuer) => {
-					const supportedTypes = issuer.metadata['response_types_supported'] as string[] | undefined;
+		this.client = null;
 
-					if (!supportedTypes?.includes('code')) {
-						logger.error('OpenID provider does not support required code flow');
+		// preload client
+		this.getClient().catch((e) => {
+			logger.error(e, '[OpenID] Failed to fetch provider config');
 
-						reject(
-							new InvalidProviderConfigError({
-								provider: additionalConfig['provider'],
-							}),
-						);
-					}
+			if (issuerDiscoveryMustSucceed !== false) {
+				logger.error(
+					`AUTH_${provider.toUpperCase()}_ISSUER_DISCOVERY_MUST_SUCCEED is enabled and discovery failed, exiting`,
+				);
+
+				process.exit(1);
+			}
+		});
+	}
+
+	private async getClient() {
+		if (this.client) return this.client;
```

Comment:
A downside to this approach is a possible race-condition that occurs when 2 calls take place, the second one being called before the first one resolves its issuer. In this case we will end up calling issuer twice.